Some unit tests failing

Probably as a result of changes in xarray behaviour, three unit tests are currently failing in CI runs:

```
FAILED tests/test_converter.py::MainTest::test_dry_run_with_higher_verbosity - FileNotFoundError: No such file or directory: '<fsspec.mapping.FSMap object at 0x13265f9b0>'
FAILED tests/test_dataslice.py::DatasliceTest::test_find_slice_no_store - FileNotFoundError: No such file or directory: '/this/path/does/not/exist'
FAILED tests/test_preprocessor.py::DatasetPreProcessorTest::test_time_var_is_scalar - ValueError: dimension 'time' already exists as a scalar variable
```

--- a/appveyor.yml
+++ b/appveyor.yml
@@ -2,7 +2,7 @@ version: '{build}'
 
 image:
   - macos
-  - Ubuntu2004
+  - Ubuntu2204
 
 configuration: Release
 
@@ -20,7 +20,7 @@ for:
 -
   matrix:
     only:
-      - image: Ubuntu2004
+      - image: Ubuntu2204
   install:
     - wget -qO- https://micromamba.snakepit.net/api/micromamba/linux-64/latest | tar -xvj bin/micromamba --strip-components=1
 

--- a/appveyor.yml
+++ b/appveyor.yml
@@ -27,7 +27,7 @@ for:
 build_script:
   - ./micromamba shell init -s bash -p ~/micromamba
   - source ~/.bashrc
-  - source ~/.bash_profile
+  - source ~/.profile
   - hash -r
   - mkdir -p ~/micromamba/pkgs/
   - export MAMBA_ROOT_PREFIX=~/micromamba

--- a/appveyor.yml
+++ b/appveyor.yml
@@ -15,14 +15,15 @@ for:
     only:
       - image: macos
   install:
-    - curl -Ls https://micromamba.snakepit.net/api/micromamba/osx-64/latest | tar -xvj bin/micromamba
+    - curl -Ls https://micro.mamba.pm/api/micromamba/osx-64/latest | tar -xvj bin/micromamba
     - mv bin/micromamba ./micromamba
 -
   matrix:
     only:
       - image: Ubuntu2204
   install:
-    - wget -qO- https://micromamba.snakepit.net/api/micromamba/linux-64/latest | tar -xvj bin/micromamba --strip-components=1
+    - curl -Ls https://micro.mamba.pm/api/micromamba/linux-64/latest | tar -xvj bin/micromamba
+    - mv bin/micromamba ./micromamba
 
 build_script:
   - ./micromamba shell init -s bash -p ~/micromamba

--- a/nc2zarr/dataslice.py
+++ b/nc2zarr/dataslice.py
@@ -60,7 +60,7 @@ def find_slice(store: Union[str, MutableMapping],
     """
     try:
         cube = xr.open_dataset(store, engine="zarr")
-    except (GroupNotFoundError, PathNotFoundError):
+    except (GroupNotFoundError, PathNotFoundError, FileNotFoundError):
         # zarr directory does not exist
         return -1, 'create'
 

--- a/nc2zarr/writer.py
+++ b/nc2zarr/writer.py
@@ -252,13 +252,15 @@ class DatasetWriter:
         output_ds = None
         try:
             output_ds = xr.open_zarr(self._output_store)
-        except GroupNotFoundError:
+        except (GroupNotFoundError, FileNotFoundError):
             pass
             # Output store doesn't exist, so we'll skip checks on it.
             # (This should only happen in the case of a dry run, in cases
             # where the output file would have been created in a normal
             # run, so self._output_path_exists is true but the Zarr doesn't
-            # actually exist.)
+            # actually exist.) xarray raises FileNotFoundError since
+            # v2022.09.0; older versions raise GroupNotFoundError (see
+            # xarray issue 6484).
         if self._output_append_mode is not AppendMode.all:
             if not self._is_append_dim_monotonic_increasing(output_ds):
                 raise ValueError(

--- a/tests/test_preprocessor.py
+++ b/tests/test_preprocessor.py
@@ -75,8 +75,10 @@ class DatasetPreProcessorTest(unittest.TestCase):
         time = ds.time
         ds = ds.rename_vars({'time': 't'})
         ds = ds.swap_dims({'time': 't'})
-        ds = ds.assign_coords({'time': xr.DataArray(time[0].values.item(),
-                                                    dims=())})
+        ds = ds.assign_coords(
+            {'time_tmp': (xr.DataArray(time[0].values.item(), dims=()))}
+        )
+        ds = ds.rename_vars({'time_tmp': 'time'})
         ds.time.attrs.update(time.attrs)
         ds.time.encoding.update(time.encoding)
         ds = ds.drop_vars('t')

